src/main/java/org/example/jvspringbootfirstbook/service/user/UserServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package org.example.jvspringbootfirstbook.service.user;
+
+import lombok.RequiredArgsConstructor;
+import org.example.jvspringbootfirstbook.dto.UserRegistrationRequestDto;
+import org.example.jvspringbootfirstbook.dto.UserResponseDto;
+import org.example.jvspringbootfirstbook.exception.RegistrationException;
+import org.example.jvspringbootfirstbook.mapper.UserMapper;
+import org.example.jvspringbootfirstbook.repository.user.UserRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto request) {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
```

Comment:
Think if user shouldn't be saved to a db here.
` @Override
    public UserResponseDto register(UserRegistrationRequestDto request) {
        validateRequest(request);
        checkIfUserExists(request.getEmail());
        User user = userMapper.toModel(request);
        User savedUser = userRepository.save(user);
        return userMapper.toDto(savedUser);
    }`